### PR TITLE
LaVague QA CLI

### PR DESCRIPTION
**Pull request by paulpalmieri**
- new CLI `lavague-qa`
    - generates pytest files from gherkin test cases (with or without LLMs)
    - handles contexts & token counting
- gherkin examples (4)
- associated docs
    - new side bar section with quicktour, usage, examples
    - update `index.md` (docs Home) and main repo `README.md`
- remove old examples
- update pricing utils token summary
- update `lavague-tests` context loading to handle defaults

### CODE REVIEW

File: docs/docs/lavague-qa/examples.md [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+|:---------------------------------|:--------------------------------------------------------------|:---------------------------------------------------------|
+| https://amazon.fr/               | [demo_amazon.feature](https://github.com/lavague-ai/LaVague/blob/main/lavague-qa/features/demo_amazon.feature)       | Tests the Amazon search, cart and cart deletion to ensure expected behavior |
+| https://laposte.fr/              | [demo_laposte.feature](https://github.com/lavague-ai/LaVague/blob/main/lavague-qa/features/demo_laposte.feature)     | Tests an interactive shipping cost calculator to ensure expected output |
+| https://en.wikipedia.org/        | [demo_wikipedia.feature](https://github.com/lavague-ai/LaVague/blob/main/lavague-qa/features/demo_wikipedia.feature) | Tests the login feature of Wikipedia  |
+| https://hsbc.fr/                 | [demo_hsbc.feature](https://github.com/lavague-ai/LaVague/blob/main/lavague-qa/features/demo_hsbc.feature)           | Tests proper multi-tab navigation and cookie banners |
+
+
+In each of these examples, we'll showcase how LaVague QA can be used to generate tests for critical parts of your websites.
```
Comment (lyie28) [inline docs/docs/lavague-qa/examples.md:17]: We don't really show how we used LaVague to do this. We just show the feature file and generated file. 

I would do one of the following or similar:
- Add some text about how to run the examples
- Add the command for running the examples with each example so it reads more as a guide
- Add end-to-end notebook examples with the installation, API key and then run and show files using 'cat' so users can really test it live.

File: docs/docs/lavague-qa/examples.md [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+
+!!! warning "Beta release"
+    LaVague QA is still a work in progress and may contain bugs. Join our [community of amazing contributors](https://discord.gg/invite/SDxn9KpqX9) to help make this tool more reliable!
+
+We will periodically add [examples in our repository](https://github.com/lavague-ai/LaVague/blob/main/lavague-qa/features/). In this section, we will go over each of the Gherkin files and showcase the generated code. 
+
+
+| URL                              | Link to file                                                  | Purpose                                                  |
+|:---------------------------------|:--------------------------------------------------------------|:---------------------------------------------------------|
+| https://amazon.fr/               | [demo_amazon.feature](https://github.com/lavague-ai/LaVague/blob/main/lavague-qa/features/demo_amazon.feature)       | Tests the Amazon search, cart and cart deletion to ensure expected behavior |
```
Comment (lyie28) [inline docs/docs/lavague-qa/examples.md:11]: I changed the URLs here as they were not pointing to lavague-qa/features

**Review by lyie28 — COMMENTED** [review]
Bug resolved on fresh install :)

File: docs/docs/lavague-qa/usage.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+- By default we use OpenAI `gpt-4o` as it is the best performing model so far
+- You can find [other contexts in our repository](https://github.com/lavague-ai/LaVague/tree/main/lavague-tests/contexts)
+
+To run with a custom context, use the `--context` flag along with the path to the `.py` file creating the objects.
+
+```bash
+lavague-qa --context ./my_contexts/custom_context_gemini.py
+```
+
+### Flag `--full-llm` or `-llm`
```
Comment (lyie28) [inline docs/docs/lavague-qa/usage.md:113]: This is not clear for me. Can you re-review this and make sure it's clear what happens for full-llm and what happens for llm & by default which (if either) is used

**Review by lyie28 — CHANGES_REQUESTED** [review]
- In examples, add command for how to run the examples
- Make the --full-llm --llm flag section a little clearer
- Briefly mention headless/non-headless option in QT as quite important
- Remove no_llm suffix for deafult - only have suffix when --full-llm flag used

File: docs/docs/lavague-qa/usage.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+```
+Usage: lavague-qa [OPTIONS]
+
+Options:
+  -u, --url TEXT      URL of the site to test
+  -f, --feature TEXT  Path to the .feature file containing Gherkin
+  -llm, --full-llm    Enable full LLM pytest generation
+  -c, --context TEXT  Path of python file containing an initialized context
+                      and token_counter. Defaults to OpenAI GPT4o
+  -h, --headless      Enable headless mode for the browser
```
Comment (lyie28) [inline docs/docs/lavague-qa/usage.md:52]: I think it's worth adding a line below explaining what this means as it's a very important option + add a quick note in the QT that you can run with headless or non-headless mode

**Review by lyie28 — APPROVED** [review]

**Review by adeprez — APPROVED** [review]

**Pull request merged**